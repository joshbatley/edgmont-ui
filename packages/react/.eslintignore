**/lib/*
**/node_modules/
**/jest.config.js
**/setupTests.ts
**/rollup.config.js
**/dist/*
**/src/tailwind.css
**/docs/*
**/preset.js
