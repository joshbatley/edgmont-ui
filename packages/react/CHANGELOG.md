# @edgmont-ui/react

## 1.0.0

### Major Changes

- b4b59a8: Revamped all components, add themes and fix broken logix

## 0.3.0

### Minor Changes

- cc0ffd7: Updated readme and version

## 3.0.1

### Patch Changes

- 1c12ec6: readme

## 3.0.0

### Major Changes

- 52e9e42: asdas

## 2.0.2

### Patch Changes

- 7e13337: asd

## 2.0.1

### Patch Changes

- cba70fc: asdada

## 2.0.0

### Major Changes

- 323e843: ad

### Minor Changes

- a80937b: asd

## 1.0.0

### Major Changes

- c4b8b3f: asd
