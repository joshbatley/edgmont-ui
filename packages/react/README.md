<p align="center">
  <img src="/public/logo-dark.svg#gh-dark-mode-only" height="128"/>
  <img src="/public/logo-light.svg#gh-light-mode-only" height="128"/>
</p>
<div align="center">

[![Build][build-image]][build-url] [![NPM Version][version-image]][version-url] [![License][license-image]][license-url] [![Bundle Size][size-image]][size-url]

[build-image]: https://img.shields.io/github/actions/workflow/status/joshbatley/edgmont-ui/npm-publish.yml
[build-url]: https://github.com/joshbatley/edgmont-ui/actions
[version-image]: https://img.shields.io/npm/v/@edgmont-ui/react
[version-url]: https://www.npmjs.com/package/@edgmont-ui/react
[license-image]: https://img.shields.io/github/license/joshbatley/@edgmont-ui/react
[license-url]: https://github.com/joshbatley/edgmont-ui/blob/main/LICENSE
[size-image]: https://img.shields.io/bundlephobia/minzip/@edgmont-ui/react
[size-url]: https://bundlephobia.com/package/@edgmont-ui/react

</div>

## Getting Started

### Installation methods

There are a few methods to choose from depending on your usage for edgmont-ui

##### Install tailwind

Follow this official set-up [guide](https://tailwindcss.com/docs/guides/create-react-app) to get tailwind working within your react application.

##### Install edgmont-ui

```
npm/yarn install @edgmont-ui/react
```

## Package used

- React - Framework of choice
- Typescript - Types!!!
- Heroicons - for the few icons needed
- Downshift - a easy to use API for our select element
- React-datepicker - for the datepicker api
- Date-fns - the datepicker date library of choice
- Prism-react-renderer & react-simple-code-editor - for the code editor and syntax highlighter
- React-dropzone - for the file uploader
- React-spring - for any animation/transitions
- React-popper - for tooltip and popover API
- Storybooks - for the docs

## License

This library is MIT licensed.