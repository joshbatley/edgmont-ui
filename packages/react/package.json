{
  "name": "@edgmont-ui/react",
  "version": "1.0.2",
  "private": false,
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "homepage": "https://ui.edgmont.dev/",
  "bugs": {
    "url": "https://github.com/joshbatley/edgmont-ui/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/joshbatley/edgmont-ui"
  },
  "jest": {
    "preset": "@repo/jest-presets/jest/browser"
  },
  "contributors": [
    "joshbatley"
  ],
  "files": [
    "dist",
    "package.json"
  ],
  "scripts": {
    "build": "tsup src/index.ts --format esm,cjs --dts --external react",
    "dev": "tsup src/index.ts --format esm,cjs --watch --dts --external react",
    "lint": "eslint . --max-warnings 0",
    "test": "jest",
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist"
  },
  "devDependencies": {
    "@repo/eslint-config": "workspace:*",
    "@repo/jest-presets": "workspace:*",
    "@repo/typescript-config": "workspace:*",
    "@turbo/gen": "^1.13.3",
    "@types/eslint": "^8.56.10",
    "@types/node": "^20.12.12",
    "@types/react": "^18.3.2",
    "@types/react-dom": "^18.3.0",
    "eslint": "^8.57.0",
    "react": "^18.3.1",
    "typescript": "5.4.5"
  },
  "dependencies": {
    "@heroicons/react": "^2.1.3",
    "@popperjs/core": "^2.11.8",
    "@testing-library/jest-dom": "^6.4.5",
    "@testing-library/react": "^14.3.1",
    "@testing-library/react-hooks": "^8.0.1",
    "@testing-library/user-event": "^14.5.2",
    "@types/jest": "^29.5.12",
    "@types/react-datepicker": "^6.2.0",
    "@types/styled-components": "^5.1.34",
    "@types/styled-system": "^5.1.22",
    "date-fns": "^3.6.0",
    "downshift": "^8.5.0",
    "jest": "^29.7.0",
    "jest-styled-components": "^7.2.0",
    "polished": "^4.3.1",
    "prism-react-renderer": "^2.3.1",
    "react-datepicker": "^6.9.0",
    "react-dom": "^18.3.1",
    "react-dropzone": "^14.2.3",
    "react-focus-lock": "^2.12.1",
    "react-popper": "^2.3.0",
    "react-simple-code-editor": "^0.13.1",
    "react-spring": "^9.7.3",
    "react-use": "^17.5.0",
    "styled-components": "^5.3.11",
    "styled-system": "^5.1.5",
    "tsup": "^8.0.2"
  }
}