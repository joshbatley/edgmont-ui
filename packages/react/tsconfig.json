{
  "extends": "@edgmont-ui/typescript-config/react-library.json",
  "include": ["."],
  "exclude": ["dist", "build", "node_modules"]
}
