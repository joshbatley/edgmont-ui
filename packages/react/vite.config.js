import react from '@vitejs/plugin-react';
import path from 'node:path';
import { defineConfig } from 'vite';
import dts from 'vite-plugin-dts';

export default defineConfig({
  plugins: [
    react(),
    dts({
      insertTypesEntry: true,
    }),
  ],
  build: {
    copyPublicDir: false,
    sourcemap: true,
    lib: {
      entry: path.resolve(__dirname, "src/index.ts"),
      name: 'index',
      formats: ['es', 'umd'],
      fileName: (format) => `index.${format}.js`,
    },
    rollupOptions: {
      external: ['react', 'react-dom', 'styled-components'],
      output: {
        globals: {
          react: 'React',
          'react-dom': 'ReactDOM',
          'styled-components': 'styled',
        },
      },
    },
  },
});
