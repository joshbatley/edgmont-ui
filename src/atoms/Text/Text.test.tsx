import React, { ReactNode } from 'react';
import { render as testRender } from '@testing-library/react';
import { Text, TextMono } from './';

function render(component: ReactNode) {
  return testRender(<>{component}</>);
}

describe('Text component', () => {
  describe('Text', () => {
    it('should render', () => {
      const { container } = render(<Text>Text</Text>);
      expect(container.firstChild).not.toBeEmptyDOMElement();
      expect(container.firstChild).toHaveTextContent('Text');
    });

    it('should match snapshots', () => {
      const { container } = render(<Text>Text</Text>);
      expect(container).toMatchSnapshot();
    });

    it('should default to <p> html tag', () => {
      const { container } = render(<Text>Text</Text>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should updated to span html tag', () => {
      const { container } = render(<Text as="span">Text</Text>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('TextMono', () => {
    it('should render', () => {
      const { container } = render(<TextMono>Text</TextMono>);
      expect(container.firstChild).not.toBeEmptyDOMElement();
      expect(container.firstChild).toHaveTextContent('Text');
    });

    it('should match snapshots', () => {
      const { container } = render(<TextMono>Text</TextMono>);
      expect(container).toMatchSnapshot();
    });
  });
});
