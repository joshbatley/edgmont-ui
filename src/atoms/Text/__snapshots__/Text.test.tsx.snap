// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Text should default to <p> html tag 1`] = `
<p>
  Text
</p>
`;

exports[`Text component Text should match snapshots 1`] = `
<div>
  <p>
    Text
  </p>
</div>
`;

exports[`Text component Text should updated to span html tag 1`] = `
<span>
  Text
</span>
`;

exports[`Text component TextMono should match snapshots 1`] = `
<div>
  <p
    class="font-mono"
  >
    Text
  </p>
</div>
`;
